### ISSUE TITLE
[lib/convert] Remove `Utf8Decoder._convertIntercepted`

### ISSUE THREAD

**Issue by rakudrama** [dart-lang/sdk#54018]
The hook `Utf8Decoder._convertIntercepted` is at the wrong level in in the UTF-8 decoder stack. It is not used on the chunked path. It does not allow a input to be converted to a `Uint8List` for the purpose of the hook, and then have that work reused if hook ultimately cannot complete the conversion.

Each platform already must provide a patch for `_Utf8Decoder.convertSingle` and `_Utf8Decoder.convertChunked`. It would be better for all platforms to do any platform specific work in these methods.

[This CL](https://dart-review.googlesource.com/c/sdk/+/335660) makes the change suggested here in js_runtime, which will also avoid the most of the problem mentioned in b/308643579. A similar change in the js_dev_runtime will be required.

/cc @osa1

**Comment by osa1** [comment]
Can we remove `_convertIntercepted`? All `Utf8Decoder.convert` does is calling backend-specific functions:

```dart
  String convert(List<int> codeUnits, [int start = 0, int? end]) {
    var result = _convertIntercepted(_allowMalformed, codeUnits, start, end); // <--- external _convertIntercepted
    if (result != null) {
      return result;
    }
    return _Utf8Decoder(_allowMalformed).convertSingle(codeUnits, start, end); // <-- external convertSingle
  }
```

So in a sense the conversion is always intercepted anyway, I don't understand why there needs to be a check for it in the shared part of the standard library.

We could simplify it as the following and let the backends do any conversions if they want:

```dart
  String convert(List<int> codeUnits, [int start = 0, int? end]) =>
      _Utf8Decoder(_allowMalformed).convertSingle(codeUnits, start, end);
```

I think `_convertIntercepted` is just making things confusing for no benefit, and probably the backends today should just return `null` from `_convertIntercepted` always.

**Comment by rakudrama** [comment]
> Can we remove `_convertIntercepted`? All `Utf8Decoder.convert` does is calling backend-specific functions:

Yes, that is the goal here. As you note, each platform can do its own special thing in `_Utf8Decoder.convertSingle`.

We can remove `_convertIntercepted` when all platforms have a trivial `return null;` implementation.
